server/chat-commands.js

```diff
@@ -4156,39 +4156,9 @@ const commands = {
 		const forPunishment = target === 'forpunishment';
 		if (forPunishment && !this.can('lock')) return false;
 
-		const battle = room.battle;
-		// retrieve spectator log (0) if there are privacy concerns
-		const format = Dex.getFormat(room.format, true);
-
-		// custom games always show full details
-		// random-team battles show full details if the battle is ended
-		// otherwise, don't show full details
-		let hideDetails = !format.id.includes('customgame');
-		if (format.team && battle.ended) hideDetails = false;
-
-		const data = room.getLog(hideDetails ? 0 : -1);
-		const datahash = crypto.createHash('md5').update(data.replace(/[^(\x20-\x7F)]+/g, '')).digest('hex');
-		let rating = 0;
-		if (battle.ended && room.rated) rating = room.rated;
-		const [success] = await LoginServer.request('prepreplay', {
-			id: room.id.substr(7),
-			loghash: datahash,
-			p1: battle.p1.name,
-			p2: battle.p2.name,
-			format: format.id,
-			rating: rating,
-			hidden: forPunishment || room.unlistReplay ? '2' : room.isPrivate || room.hideReplay ? '1' : '',
-			inputlog: battle.inputLog ? battle.inputLog.join('\n') : null,
-		});
-		if (success) battle.replaySaved = true;
-		if (success && success.errorip) {
-			connection.popup(`This server's request IP ${success.errorip} is not a registered server.`);
-			return;
-		}
-		connection.send('|queryresponse|savereplay|' + JSON.stringify({
-			log: data,
-			id: room.id.substr(7),
-		}));
+		const result = await room.saveReplay(forPunishment);
+		if (result.error) return this.popupReply(result.error);
+		connection.send('|queryresponse|savereplay|' + JSON.stringify(result));
```

Comment:
```suggestion
		connection.send(`|queryresponse|savereplay|` + JSON.stringify(result));
```

Reply:
Should we convert this all the way? `` `|queryresponse|savereplay|${JSON.stringify(result)` ``?

Reply:
No strong opinion. String standards just say `` ` `` for protocol code.